### ISSUE TITLE
Using 12V lead acid battery 

### ISSUE THREAD

**Issue by js333031** [EFeru/hoverboard-firmware-hack-FOC#328]
### Description

I don't have the 36V Li Ion batteries but do have the mainboard and motors/wheels. Is it possible to:
1. Flash new firmware w/o power source connected to the XT60 connector? The firmware flashing instructions seem to allow so but older documentation I believe required ">36V" power connected to XT60
2. Run the firmware and motors w/12V lead acid battery?

I happen to have the Pb battery and wanted to check before I purchase Li battery.

**Comment by Candas1** [comment]
Hi,

You can power the chip with 3.3v from stlink, but it's not recommended, it seems few people ended up bricking the board.
I am not sure running on 12v is a good idea, the power button will for sure not work with that low voltage.

**Comment by RoboDurden** [comment]
The mosfet gate drivers run on 14V. but technically they should already work at 12V.

**Comment by derFrickler** [comment]
Is there a way to enable the board without the power button? just turning on when plugging power?
Want to try to run the board on a 4S lipo

**Comment by RoboDurden** [comment]
Shorting the first TIP127 TO220 transistor will always power the board. This happens when that transistor blows. These boards then will always turn on once the battery is connected.
Have a look at the schematic.pdf ..

You should uncomment 
```
  if (inactivity_timeout_counter > (INACTIVITY_TIMEOUT * 60 * 1000) / (DELAY_IN_MAIN_LOOP + 1)) {  // rest of main loop needs maybe 1ms
      poweroff();
    }
```
in main.c because the power down melody might hang at the last tone :-(

**Comment by derFrickler** [comment]
Great, thanks a lot, I will try that

**Comment by derFrickler** [comment]
Seems its not an TIP127 on my board (the 2nd on is), first one is an IRF9530N.
But bridging source and drain did the Job ;_)

Thanks a lot!

**Comment by RoboDurden** [comment]
Yes, some boards have a mosfet instead of a to220 transistor. Not really neccessary. Don't forget to post a youtube video when you have something nice to show :-)

**Comment by abyy28y** [comment]
does it work for u derfrickler?

**Comment by derFrickler** [comment]
YES, works great, i will do a video soon.

**Comment by abyy28y** [comment]
works on 12 volt?

**Comment by abyy28y** [comment]
> Shorting the first TIP127 TO220 transistor will always power the board. This happens when that transistor blows. These boards then will always turn on once the battery is connected. Have a look at the schematic.pdf ..
> 
> You should uncomment
> 
> ```
>   if (inactivity_timeout_counter > (INACTIVITY_TIMEOUT * 60 * 1000) / (DELAY_IN_MAIN_LOOP + 1)) {  // rest of main loop needs maybe 1ms
>       poweroff();
>     }
> ```
> 
> in main.c because the power down melody might hang at the last tone :-(

excuse me is there is any way or tip127 substitute to make the board powered on 12 volt using the power button?

**Comment by RoboDurden** [comment]
> excuse me is there is any way or tip127 substitute to make the board powered on 12 volt using the power button?

The power button triggers the littler transistor Q1 which will activate that first TIP127.
You might need to decrease the resistance of R1 from 30 kOhm to maybe 20k or 10k in order to trigger Q1 when V_bat is only 12V instead of 36V.
See the schematics pages 5 and 6 of https://raw.githubusercontent.com/EFeru/hoverboard-firmware-hack-FOC/main/docs/20150722_hoverboard_sch.pdf

You could try to feed in the 12V with a 10k Ohm resistor directly into the On_Switch Jumper J1 pin 1..
Just a quick idea..

I am currently working on the Gen2 boards: https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x

**Comment by abyy28y** [comment]
Thank u so much RoboDurden and good luck, I will try it.

**Comment by ash25mi** [comment]
Hi, any news on using a 12V battery ?

**Comment by derFrickler** [comment]
Its working on 4s Lipo, so it should be fine at 12V aswell, just need to set the batery low levels in the code.

**Comment by derFrickler** [comment]
BTW, i promised a  video, heres one of the first test-drives: 
https://www.youtube.com/watch?v=epiRJ1xkm4k

**Comment by derFrickler** [comment]
I bought the tracks years ago from Aliexpress for $40, but they are not available anymore (at least not from Europe).
Power is fine like this as its only a fun vehicle and i am only using 4S right now. It was able to pull the Sled with my 2 small kids on the Snow...

**Comment by RoboDurden** [comment]
I don't like fun projects... 
Better make it a $200 12s suicide gps bomber tank to support Ukrainian soldiers :-/

**Comment by ash25mi** [comment]
> Its working on 4s Lipo, so it should be fine at 12V aswell, just need to set the batery low levels in the code.
Thanks for the feedback ! Nice build.

> I forgot to tell you that I auto-power-on my solar camper with a 10 uF Capacitor and a ? KOhm discharge resistor. When powering the board, the capacitor is discharged and simulates a short button push :-) That way you eliminate the big spark when plugging in the battery. Even 10A Power switches do not last long because of the big capacitors of the hoverboard controller that I only use to generate up to 2 amps pedal assist.
Thanks it is a good addition !

> I don't like fun projects... Better make it a $200 12s suicide gps bomber tank to support Ukrainian soldiers :-/
Small drones in Ukraine are also consumer grade, sometimes based on open-source projects, and also using hobby components.
If you want to see it that way, the existence of hobbies enables people to learn skills and create projects, and an industry to exist.

**Comment by ash25mi** [comment]
> Its working on 4s Lipo, so it should be fine at 12V aswell, just need to set the batery low levels in the code.
Thanks for the feedback ! Nice build.

> I forgot to tell you that I auto-power-on my solar camper with a 10 uF Capacitor and a ? KOhm discharge resistor. When powering the board, the capacitor is discharged and simulates a short button push :-) That way you eliminate the big spark when plugging in the battery. Even 10A Power switches do not last long because of the big capacitors of the hoverboard controller that I only use to generate up to 2 amps pedal assist.

Thanks it is a good addition !

> I don't like fun projects... Better make it a $200 12s suicide gps bomber tank to support Ukrainian soldiers :-/

Small drones in Ukraine are also consumer grade, sometimes based on open-source projects, and also using hobby components.
If you want to see it that way, the existence of hobbies enables people to learn skills and create projects, and an industry to exist.

**Comment by ash25mi** [comment]
> Its working on 4s Lipo, so it should be fine at 12V aswell, just need to set the batery low levels in the code.

Thanks for the feedback ! Nice build.

> I forgot to tell you that I auto-power-on my solar camper with a 10 uF Capacitor and a ? KOhm discharge resistor. When powering the board, the capacitor is discharged and simulates a short button push :-) That way you eliminate the big spark when plugging in the battery. Even 10A Power switches do not last long because of the big capacitors of the hoverboard controller that I only use to generate up to 2 amps pedal assist.

Thanks it is a good addition !

> I don't like fun projects... Better make it a $200 12s suicide gps bomber tank to support Ukrainian soldiers :-/

Small drones in Ukraine are also consumer grade, sometimes based on open-source projects, and also using hobby components.
If you want to see it that way, the existence of hobbies enables people to learn skills and create projects, and an industry to exist.

**Comment by TinkerWorX** [comment]
Just wanted to share a quick example for anyone looking how to quickly make the board turn off with lower voltage. For my setup I wanted it to run off 18v Ryobi batteries which wasn't letting it power. I added a 22k ohm resistor as shown in the image. This drops the R1 resistance from 30k to 12.6k. (ignore the soldering job, hard to apply enough heat on the massive copper area)
![image](https://github.com/EFeru/hoverboard-firmware-hack-FOC/assets/1730988/6a49ed38-6c27-44d1-8ff9-3012b4707d0b)
It now powers up from 18v without issues. Remember to update the firmware as well to avoid it shutting down from under-voltage detection.

**Comment by RoboDurden** [comment]
Be aware that now attaching a fully loaded 36V battery might exceed the 3.3V/5V of the MCU input pin !

**Comment by TinkerWorX** [comment]
Good point, I added a warning to my post to make sure people understand this.

**Comment by thecesco** [comment]
> Shorting the first TIP127 TO220 transistor will always power the board. This happens when that transistor blows. These boards then will always turn on once the battery is connected. Have a look at the schematic.pdf ..
> 
> You should uncomment
> 
> ```
>   if (inactivity_timeout_counter > (INACTIVITY_TIMEOUT * 60 * 1000) / (DELAY_IN_MAIN_LOOP + 1)) {  // rest of main loop needs maybe 1ms
>       poweroff();
>     }
> ```
> 
> in main.c because the power down melody might hang at the last tone :-(

Hi! Could you please help how to short the tip? Thanks!

**Comment by RoboDurden** [comment]
If you don't know how to google and read the TIP127 datasheet, you should not shortcut any electronics.

**Comment by derFrickler** [comment]
As stated above, i shorted the TIP127/IRF9530N next to the switch and commented the poweroff line.
That worked fine for me.

BTW, thats where the tracked platform ended up ;_)
https://der-frickler.net/spass/hoverbob

On the page you find some of my other hoverboard projects, like the 4wd quad i did lately.
